src/ucp/wireup/wireup.c

```diff
@@ -1525,7 +1588,11 @@ ucp_wireup_check_config_intersect(ucp_ep_h ep, ucp_ep_config_key_t *new_key,
                     "new_key->wireup_msg_lane=%u", new_key->wireup_msg_lane);
     }
 
-    wireup_lane = ucp_wireup_get_msg_lane(ep, UCP_WIREUP_MSG_REQUEST);
+    am_need_flush = ucp_wireup_is_am_need_flush(ep, reuse_lane_map);
+    wireup_lane   = am_need_flush ? ucp_ep_get_am_lane(ep) :
+                            /* EP reconfiguration can only occur after wireup
+                             * request was sent (thus UCP_WIREUP_MSG_REQUEST) */
+                            ucp_wireup_get_msg_lane(ep, UCP_WIREUP_MSG_REQUEST);
```

Comment:
are we going to [re-]send wireup request in every occurrence of `ucp_wireup_get_msg_lane(ep, UCP_WIREUP_MSG_REQUEST)` in the PR? if so, maybe add a helper function with clear purpose?